### PR TITLE
HBASE-23203 NPE in RSGroup info

### PR DESCRIPTION
**Pull request by karthikhw**
Rsgroup.jsp call Admin#listTableDescriptors((Pattern)null, true) with Pattern null but implementation RawAsyncHBaseAdmin#listTableDescriptors don't allow null by Precondition.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   6m 18s |  master passed  |
| :green_heart: |  compile  |   1m 56s |  master passed  |
| :green_heart: |  checkstyle  |   2m 41s |  master passed  |
| :green_heart: |  shadedjars  |   5m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 29s |  master passed  |
| :blue_heart: |  spotbugs  |   1m 35s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   7m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m 30s |  the patch passed  |
| :green_heart: |  compile  |   2m  0s |  the patch passed  |
| :green_heart: |  javac  |   2m  0s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 40s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m  0s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  17m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 28s |  the patch passed  |
| :broken_heart: |  findbugs  |   1m 46s |  hbase-thrift generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   1m 51s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 254m 28s |  hbase-server in the patch failed.  |
| :green_heart: |  unit  |   5m 50s |  hbase-thrift in the patch passed.  |
| :green_heart: |  asflicense  |   1m 21s |  The patch does not generate ASF License warnings.  |
|  |   | 339m 19s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hbase-thrift |
|  |  There is an apparent infinite recursive loop in org.apache.hadoop.hbase.thrift2.client.ThriftAdmin.listTableDescriptors(boolean)  At ThriftAdmin.java:recursive loop in org.apache.hadoop.hbase.thrift2.client.ThriftAdmin.listTableDescriptors(boolean)  At ThriftAdmin.java:[line 144] |
| Failed junit tests | hadoop.hbase.master.TestAssignmentManagerMetrics |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/747 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux e75eb08f410a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-747/out/precommit/personality/provided.sh |
| git revision | master / 2ad62b0162 |
| Default Java | 1.8.0_181 |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/1/artifact/out/new-findbugs-hbase-thrift.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/1/testReport/ |
| Max. process+thread count | 4860 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-thrift U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java [hunk 1]
```diff
@@ -124,6 +124,14 @@
    */
   List<TableDescriptor> listTableDescriptors() throws IOException;
 
+  /**
+   * List all tables including system tables.
```
[the remark this answers is not in the document]
Comment (karthikhw) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java]: Thank you very much @Apache9 (Duo Zhang) for reviewing the change. Corrected it.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   1m  1s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   6m 24s |  master passed  |
| :green_heart: |  compile  |   2m 10s |  master passed  |
| :green_heart: |  checkstyle  |   3m 14s |  master passed  |
| :green_heart: |  shadedjars  |   5m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 52s |  master passed  |
| :blue_heart: |  spotbugs  |   1m 51s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   8m 19s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   7m 33s |  the patch passed  |
| :green_heart: |  compile  |   2m 27s |  the patch passed  |
| :green_heart: |  javac  |   2m 27s |  the patch passed  |
| :green_heart: |  checkstyle  |   3m  3s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  21m 37s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 53s |  the patch passed  |
| :green_heart: |  findbugs  |   9m 15s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   2m 13s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 245m 38s |  hbase-server in the patch failed.  |
| :green_heart: |  unit  |   5m 31s |  hbase-thrift in the patch passed.  |
| :green_heart: |  asflicense  |   1m 30s |  The patch does not generate ASF License warnings.  |
|  |   | 345m 56s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/747 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 93bc3d42dc5a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-747/out/precommit/personality/provided.sh |
| git revision | master / 59fc163e94 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/2/testReport/ |
| Max. process+thread count | 4865 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-thrift U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-747/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
The failed UTs are not related. Let me merge.

**Pull request merged**